clients/gateway.go

```diff
@@ -210,9 +233,39 @@ type Abi []struct {
 		Name string `json:"name"`
 		Type string `json:"type"`
 	} `json:"inputs"`
-	Outputs []interface{} `json:"outputs"`
+	Outputs []struct {
+		Name string `json:"name"`
+		Type string `json:"type"`
+	} `json:"outputs"`
+	StateMutability string `json:"stateMutability"`
+	Data            []struct {
+		Name string `json:"name"`
+		Type string `json:"type"`
+	}
+	Keys []interface{} `json:"keys"`
```

Comment:
thanks. i didnt know about this

Reply:
You're welcome.

We should do the same thing for the `Program` struct. Then, we can change `Hints` from `interface{}` type to `map[uint64]interface{}` type.

Reply:
yea you are right again! how about map[uint64][]interface{}?

Reply:
Can't completely recall. But I think the last time I marshalled using `map[uint64][]interface{}`, json sometimes rearranged the array elements. But we don't want that to happen because the original order is very important when computing hashes.

Reply:
oh yea ok gotcha. anyway i have updated it